client/quantum_serverless/core/job.py

```diff
@@ -496,7 +496,7 @@ def result(self, wait=True, cadence=5, verbose=False):
         if wait:
             if verbose:
                 logging.info("Waiting for job result.")
-            while not self._in_terminal_state():
+            while not self.in_terminal_state():
```

Comment:
We don't have a `JobStatus` object like ray that we can query like Ray's `JobStatus.is_terminal`, so we will make this `Job.in_terminal_state` method public, so clients have a way to query whether the job has finished.